### ISSUE TITLE
Add evaluate function to AutoModel

### ISSUE THREAD

**Issue by jhfjhfj1** [keras-team/autokeras#705]
Would it be relevant to have an `evaluate` method as well? e.g. a standard practice I would expect is to have 2 validation sets, one used for search during training, and one used at the end as the actual validation set (to prevent validation overfitting).

_Originally posted by @fchollet in https://github.com/keras-team/autokeras/pull/691_
